src/pages/PasswordResetPage.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import { motion } from 'framer-motion';
+import { useHistory } from 'react-router-dom';
+import { notification } from 'antd';
+import PropTypes from 'prop-types';
+import Card from '../components/Card.jsx';
+import TextInput from '../components/TextInput.jsx';
+import Button from '../components/Button.jsx';
+import '../scss/password-reset-page.scss';
+import API from '../api/API';
+
+const animationVariants = {
+  hidden: {
+    opacity: 0,
+    transform: 'translateY(50%)',
+  },
+  show: {
+    opacity: 1,
+    transform: 'translateY(0%)',
+  },
+};
+
+const animationOpts = {
+  duration: 1.5,
+  ease: [0.16, 1, 0.3, 1],
+};
+
+function PasswordResetPage({ userId }) {
+  const params = new URLSearchParams(window.location.search);
+  const verificationCode = params.get('verificationCode');
+  const [submitted, setSubmitted] = useState(false);
+  const [password, setPassword] = useState('');
+  const [confirmedPassword, setConfirmedPassword] = useState('');
+  const history = useHistory();
+
+  function isTrimmedEmpty(str) {
+    if (str.trim() === '') return true;
+    return false;
+  }
+
+  function goBack() {
+    history.replace('/');
+    setSubmitted(false);
+  }
+
+  async function changePassword(event) {
+    event.preventDefault();
+
+    try {
+      if (!password.match(/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/g)) {
+        throw (new Error('Your password does not meet the requirements.'));
+      }
+
+      if (password !== confirmedPassword) {
+        throw (new Error('Passwords do not match.'));
+      }
```

Comment:
I think it's probably better to show notifications here and return rather than throw an error, something like
```js
if (condition) {
  notification.error({ ... });
  return;
}
```